### PR TITLE
Followup #278 Assert that quarkus-extension.json exists for each exte…

### PR DESCRIPTION
**Pull request by ppalaga**
…nsion

### CODE REVIEW

**Comment by lburgazzoli** [comment]
FWIW the file is auto generated by the quarkus plugin in any case

**Comment by ppalaga** [comment]
> FWIW the file is auto generated by the quarkus plugin in any case

My understanding is that it is generated in `${project.build.outputDirectory}` unless it exists in `src/main/resources`. But we definitely want a custom file in `src/main/resources` because of the labels and name. So we should check that we have it, should we not?

**Comment by lburgazzoli** [comment]
IMHO, the validation is meaningful if it checks that the content has some minimal info, checking only the existence of the file does not add real value

**Comment by johnpoth** [comment]
Also all these build checks are a bit painful when developing... Any chance to move them to a Profile ? Similar to what is done in Camel with the sourcecheck profile?

**Comment by ppalaga** [comment]
> IMHO, the validation is meaningful if it checks that the content has some minimal info, checking only the existence of the file does not add real value

Added some checks in 7203021

> Also all these build checks are a bit painful when developing... Any chance to move them to a Profile ? Similar to what is done in Camel with the sourcecheck profile?

I moved it to validate phase so that these issues pop up at the very end. I do not think this should be optional because failing to deliver a valid metadata is a real functional problem.

**Comment by lburgazzoli** [comment]
@ppalaga can you rebase?

**Comment by ppalaga** [comment]
2ce6dc7 rebased

File: build/scripts/validate-extension-metadata.groovy [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+final List<String> messages = []
+project.basedir.eachFile FileType.DIRECTORIES, { extensionParentDir ->
+    if (new File(extensionParentDir, 'runtime/pom.xml').exists()) {
+        final File extensionJsonFile = new File(extensionParentDir, quarkusExtensionJsonRelPath)
+        final String shortName = extensionParentDir.getName()
+        final String shortPath = shortName + '/' + quarkusExtensionJsonRelPath
+        final boolean ancillary = (shortName.startsWith('core') || shortName.endsWith('-common'))
+        if (!extensionJsonFile.exists()) {
+            messages.add(shortPath + ' is missing')
+        } else if (!ancillary) {
```
Comment (ppalaga) [inline build/scripts/validate-extension-metadata.groovy]: Actually not sure we want to ignore all these checks in the ancillary extensions. WDYT @lburgazzoli ? Maybe they should have a name, empty labels (or maybe integration, internal ?) and the guide does not matter?

Comment (lburgazzoli) [reply build/scripts/validate-extension-metadata.groovy]: core-* do already have the extension metadata file (well core-cloud has an empty one but that's a mistake) and probably xml-common should now be renamed core-xml so yes, every non support extension should have proper  metdata

Comment (ppalaga) [reply build/scripts/validate-extension-metadata.groovy]: Thanks. One more question: do we still consider the extensions satisfying `shortName.startsWith('core') || shortName.endsWith('-common')` to be "not end user facing"? - I mean should those maybe always have some sort of `internal` label that the script should enforce?

Comment (lburgazzoli) [reply build/scripts/validate-extension-metadata.groovy]: don't think so, it' perfectly acceptable for people using core alone and internal is a little misleading i.e. right now if you want to have xml support you have to explicitly add xml-common

**Comment by ppalaga** [comment]
0557eee updated based on the discussion with @lburgazzoli

File: extensions/http-common/runtime/src/main/resources/META-INF/quarkus-extension.json [hunk 2]
```diff
@@ -1 +1,8 @@
-{}
+{
+  "name": "Camel Quarkus HTTP Common",
+  "labels": [
+    "integration",
+    "camel"
```
Comment (ppalaga) [inline extensions/http-common/runtime/src/main/resources/META-INF/quarkus-extension.json]: Let me add HTTP here

**Comment by ppalaga** [comment]
0557eee Added labels to XML and HTML common

**Pull request merged**